protocol/consistencychecks.go

```diff
@@ -62,83 +61,82 @@ func NewCC(savedSTR []byte, useTBs bool, signKey sign.PublicKey) *ConsistencyChe
 	return cc
 }
 
-// Verify verifies the directory's response for a request.
-// It first verifies the directory's returned status code of the request.
+// clone makes a copy of the current consistency state.
+func (cc *ConsistencyChecks) clone() *ConsistencyChecks {
+	newCC := &ConsistencyChecks{
+		CurrentEpoch: cc.CurrentEpoch,
+		SavedSTR:     cc.SavedSTR,
+		useTBs:       cc.useTBs,
+		signKey:      cc.signKey,
+	}
+
+	if cc.useTBs {
+		newCC.TBs = make(map[string]*m.TemporaryBinding)
+		for k, v := range cc.TBs {
+			newCC.TBs[k] = v
+		}
+	}
+
+	return newCC
+}
+
+func (cc *ConsistencyChecks) update(updated *ConsistencyChecks) {
+	cc.CurrentEpoch = updated.CurrentEpoch
+	cc.SavedSTR = updated.SavedSTR
+	cc.TBs = updated.TBs
+}
+
+// Verify verifies the directory's response for a request and
+// updates the current state if all the verifications are successful.
+// It makes a copy of the current state and does verifications
```

Comment:
>  And I don't see an issue in updating the cached TBs on the fly since we only remove the TBs after they have been verified.

What should we do if, say, a TB was verified (which means it was removed from `cc.TBs`) and after that, a returned TB verifying fails (which means it should not be inserted into `cc.TBs`)? Should we discard all the verification or accept all passed checks?